### ISSUE TITLE
Implement `inverse_transform` for MultiMaskers that supports multiprocessing

### ISSUE THREAD

**Issue by arthurmensch** [nilearn/nilearn#609]
`NiftiMasker.transform_imgs` allows setting `n_jobs`. This should also be the case for `Niftimasker.inverse_transform`

**Comment by AlexandreAbraham** [comment]
Actually, `transform_imgs` should not be available in NiftiMasker since it is a MultiNiftiMasker function. The behavior of both transform and inverse_transform for these functions is subject to debate.

**Comment by banilo** [comment]
There might be advantageous scenarios in which ```NiftiMasker.transform``` might profit from (embarrasing) parallelization.

**Comment by NicolasGensollen** [comment]
@bthirion I'm wondering whether this issue is still "up-to-date". Unless I missed something, the only usage of parallelization for maskers I see is for the `MultiNiftiMasker`:

https://github.com/nilearn/nilearn/blob/3a74fcb36b17d94289b57ded5101ea9095bccc2e/nilearn/input_data/multi_nifti_masker.py#L297-L305

**Comment by bthirion** [comment]
Yes, but this is in the `transform` method, not `inverse_transform`.

**Comment by kimsin98** [comment]
I'm not sure where in `NiftiMasker.transform` parallization should take place. I'm guessing you would want it across the last dimension of 4D imgs, but going down the rabbit hole of function calls shows that most of the computation takes place in `nilearn.maskers.base_marker`'s `_filter_and_extract` function. This calls multiple functions that expect 4D imgs and/or run their own loops like `smooth_img`.

Should each of these function be parallelized? Or should I break 4D `imgs` into a list of 3D `img` earlier in the call stack then parallelize just that?

**Comment by man-shu** [comment]
Looking at the discussion, I think the title might be misleading.

We need to implement a parallelized `inverse_transform` method for the MultiNiftiMasker (not the NiftiMasker). It currently inherits the inverse_transform method from the NiftiMasker which is not parallel.

**Comment by kimsin98** [comment]
By the way, `transform_imgs` doesn't seem to be tested?

**Comment by bthirion** [comment]
> By the way, `transform_imgs` doesn't seem to be tested?

Oops, indeed, this is clearly missing.

**Comment by Remi-Gau** [comment]
> By the way, `transform_imgs` doesn't seem to be tested?

It is "tested" in at least one example but did not find any unit test

https://nilearn.github.io/stable/auto_examples/02_decoding/plot_miyawaki_encoding.html#mapping-the-encoding-scores-on-the-brain

probably should be a separate issue

**Comment by Remi-Gau** [comment]
> By the way, `transform_imgs` doesn't seem to be tested?

note that it is now tested as the MultiMixin.transform will call transforms() if dealing with a 5D image

**Comment by Remi-Gau** [comment]
I think this one could be done fairly easily by adding an inverse_transform method to the MultiMixin that can accept an iterable of 2D arrays: gonna add this one to the sprint
